modules/environments/azure_devops.tf

```diff
@@ -0,0 +1,38 @@
+resource "azuredevops_serviceendpoint_azurerm" "endpoint" {
+  count                 = var.pipeline_environment == "prod" && var.env != "ptl" && var.env != "ptlsbox" ? 1 : 0
+  project_id            = "PlatformOperations"
+  service_endpoint_name = "OPS-APPROVAL-GATE-${upper(var.env)}-ENVS"
+  credentials {
+    serviceprincipalid  = azuread_service_principal.sp.application_id
+    serviceprincipalkey = azuread_application_password.token.value
+  }
+  azurerm_spn_tenantid      = data.azurerm_client_config.current.tenant_id
+  azurerm_subscription_id   = data.azurerm_subscriptions.sds[0].subscriptions[0].subscription_id
+  azurerm_subscription_name = data.azurerm_subscriptions.sds[0].subscriptions[0].display_name
+}
+
+resource "azuredevops_serviceendpoint_azurerm" "ptl_endpoint" {
+  count                 = var.pipeline_environment == "prod" && var.env == "ptl" ? 1 : 0
+  project_id            = "PlatformOperations"
+  service_endpoint_name = "OPS-APPROVAL-GATE-${upper(var.env)}-ENVS"
+  credentials {
+    serviceprincipalid  = azuread_service_principal.sp.application_id
+    serviceprincipalkey = azuread_application_password.token.value
+  }
+  azurerm_spn_tenantid      = data.azurerm_client_config.current.tenant_id
+  azurerm_subscription_id   = data.azurerm_subscriptions.sds_ptl[0].subscriptions[1].subscription_id
```

Comment:
why is it called sds_ptl ?

Reply:
The sds ptl and ptlsbox subscriptions don't follow the DTS-SHAREDSERVICES-${ENV} naming convention so I had to split them out with their own logic

Reply:
could we add extra configuration to their subscription object (i.e. how you can override display name and environment) instead rather than extra terraform resources?